src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AbstractSuperCheck.java

```diff
@@ -72,11 +72,6 @@
     }
 
     @Override
-    public void beginTree(DetailAST rootAST) {
-        methodStack.clear();
```

Comment:
@Nimfadora You are removing this because we add on a visit and remove on a leave, right?
We talked about special circumstance with haltOnException off and an exception happening between the 2.
Please confirm this behavior and that clear is needed.

Reply:
ok, lets cover such case by reflection, comment over test method should have explanation on ability to reproduce it in testing environment(haltOnException) or incorrectly implemented executor of checkstyle.